lib/workers/repository/finalise/repository-statistics.ts

```diff
@@ -33,3 +35,55 @@ export function runRenovateRepoStats(
   }
   logger.debug({ stats: prStats }, `Renovate repository PR statistics`);
 }
+
+interface BranchMetadata {
+  branchName: string;
+  sha: string | null;
+  parentSha: string | null;
+  automerge: boolean;
+  isModified: boolean;
+}
+
+interface BaseBranchMetadata {
+  branchName: string;
+  sha: string;
+}
+
+export function runBranchSummery(): void {
+  const { scan, branches } = getCache();
+
+  const baseMetadata: BaseBranchMetadata[] = [];
+  for (const [branchName, cached] of Object.entries(scan ?? {})) {
+    baseMetadata.push({ branchName, sha: cached.sha });
+  }
+
+  function unwrap({
+    branchName,
+    sha,
+    parentSha,
+    automerge,
+    isModified,
+  }: BranchCache): BranchMetadata {
+    return { branchName, sha, parentSha, automerge, isModified };
+  }
+
+  const branchMetadata: BranchMetadata[] = [];
+  const inactiveBranches: string[] = [];
+
+  for (const branch of branches ?? []) {
```

Comment:
maybe you can add a condition before this loop to check if the array is empty.

Reply:
This is possible, but then ill have to nest the for loop as i cant return early, as i need to log the rest of the data.